compiler/damlc/daml-compiler/src/DA/Daml/Compiler/Upgrade.hs

```diff
@@ -152,6 +153,21 @@ templateInstances env externPkgId =
   where
     mod = envMod env
 
+choiceInstances :: Env -> LF.PackageId -> [HsDecl GhcPs]
+choiceInstances env externPkgId =
+    [ generateChoiceInstance env externPkgId templateDT choice
+    | template <- NM.elems $ LF.moduleTemplates mod
+    , choice <- NM.elems $ LF.tplChoices template
+    , not (specialArchiveChoice choice)
+    , Just templateDT <-
```

Comment:
Currently we will silently discard all choices if this lookup fails. Afaict if this lookup fails it’s a bug somewhere so crashing seems like a better option than silently discarding the choices.

Reply:
My defence is I was matching the style of existing code in `templateInstances`, so I assumed it was possible for this lookup to fail. But it does seem unlikely. @robin-da ?

Reply:
it's fine to drop the choice if it's not present in the dalf, but here you will drop it even if it's present but a later condition will fail. That's different from `templateInstances`, I think this case should be a crash.

Reply:
change to crash if cant find the template datatype def